File: Data Structures & Algorithms/trapping-rain-water/submission-0.cpp
class Solution {
public:
    int trap(vector<int>& height) {
        if(height.empty()){
            return 0;
        }
        int right = height.size();
        int res = 0;

        for(int left = 0; left < right; left++){
            int leftMax = height[left];
            int rightMax = height[left];

            for(int i = 0; i < left; i++){
                leftMax = max(leftMax, height[i]);
            }
            for(int i = left + 1; i < right; i++){
                rightMax = max(rightMax, height[i]);
            }

            res += min(leftMax, rightMax) - height[left];
        }
        return res;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) == 0:
            return 0

        res = 1
        l = 0
        r = 1
        seen = set(s[l])
        while r < len(s):
            while s[r] in seen:
                seen.remove(s[l])
                l += 1
            seen.add(s[r])
            res = max(res, r - l + 1)
            r += 1

        return res